modules/assim.sequential/R/Analysis_sda.R

```diff
@@ -129,6 +129,8 @@ GEF<-function(settings, Forecast, Observed, H, extraArg, nitr=50000, nburnin=100
   Pf = stats::cov(X) # Cov Forecast - Goes into tobit2space as initial condition but is re-estimated in tobit space
   mu.f <- colMeans(X) #mean Forecast - This is used as an initial condition
   
+  diag(Pf)[which(diag(Pf)==0)] <- min(diag(Pf)[which(diag(Pf) != 0)])/5 #fixing det(Pf)==0
```

Comment:
I trust you know what you're doing, but could you explain (briefly) where this is coming from? I can see why zeros on the diagonal would mess with the determinant, but why is 1/5 of the smallest diagonal element a good choice for replacing those zero elements?

Reply:
I actually got that line from the ENKF function within this file. I just added to be in the GEFS section as well since I was getting singularity issues with Pf. So I'm not sure why the 1/5 smallest diagonal was picked.